packages/iov-cli/src/helpers.spec.ts

```diff
@@ -1,16 +1,147 @@
+      const context = createContext({});
+      await executeJavaScriptAsync("var a", "myfile.js", context)
+        .then(() => fail("must not resolve"))
+        .catch(e => expect(e).toMatch(/SyntaxError:/));
+      await executeJavaScriptAsync("let b", "myfile.js", context)
+        .then(() => fail("must not resolve"))
+        .catch(e => expect(e).toMatch(/SyntaxError:/));
+      await executeJavaScriptAsync("const c", "myfile.js", context)
+        .then(() => fail("must not resolve"))
+        .catch(e => expect(e).toMatch(/SyntaxError:/));
+    });
+
+    it("can execute multiple commands in one context", async () => {
+      const context = createContext({});
+      expect(await executeJavaScriptAsync("let a = 2", "myfile.js", context)).toBeUndefined();
+      expect(await executeJavaScriptAsync("a", "myfile.js", context)).toEqual(2);
+      expect(await executeJavaScriptAsync("a += 1", "myfile.js", context)).toEqual(3);
+      expect(await executeJavaScriptAsync("a += 7", "myfile.js", context)).toEqual(10);
+      expect(await executeJavaScriptAsync("a", "myfile.js", context)).toEqual(10);
+      expect((context as any).a).toEqual(10);
+    });
+
+    it("local variables are available across multiple calls in one context", async () => {
+      const context = createContext({});
+      expect(await executeJavaScriptAsync("let a = 3", "myfile.js", context)).toBeUndefined();
+      expect(await executeJavaScriptAsync("a", "myfile.js", context)).toEqual(3);
+      expect((context as any).a).toEqual(3);
+    });
+
+    it("works with strict mode", async () => {
+      const context = createContext({});
+      expect(await executeJavaScriptAsync('"use strict"; let a = 3', "myfile.js", context)).toBeUndefined();
+      expect(await executeJavaScriptAsync('"use strict"; a', "myfile.js", context)).toEqual(3);
+      expect((context as any).a).toEqual(3);
+    });
+
+    it("can reassign const", async () => {
+      // a side-effect of local variable assignment manipulation
+      const context = createContext({});
+      expect(await executeJavaScriptAsync("const a = 3", "myfile.js", context)).toBeUndefined();
+      expect((context as any).a).toEqual(3);
+      expect(await executeJavaScriptAsync("const a = 4", "myfile.js", context)).toBeUndefined();
+      expect((context as any).a).toEqual(4);
+    });
+
+    it("can execute timeout promise code", async () => {
+      const context = createContext({ setTimeout: setTimeout });
+      const code = "await (new Promise(resolve => setTimeout(() => resolve('job done'), 5)))";
+      expect(await executeJavaScriptAsync(code, "myfile.js", context)).toEqual("job done");
+    });
+
+    it("can execute timeout code in multiple statements", async () => {
+      const context = createContext({ setTimeout: setTimeout });
+      const code = `
+        const promise = new Promise(resolve => {
+          setTimeout(() => resolve('job done'), 5);
+        });
+        await (promise);
```

Comment:
why are there parentheses around that `promise` ?

Reply:
Because the default parser used by recast needed them. Switching to babylon as a JavaScript parser makes them unnecessary. So they could be refactored away now.